src/Core/Components/List/ListComponentBase.razor.cs

```diff
@@ -364,8 +364,10 @@ protected override void OnParametersSet()
         {
             if (SelectedOption == null && Items != null && OptionSelected != null)
             {
-                TOption? item = Items.FirstOrDefault(i => OptionSelected.Invoke(i));
-                InternalValue = GetOptionValue(item);
+                var item = Items.FirstOrDefault(i => OptionSelected.Invoke(i));
+                var value = GetOptionValue(item);
+                InternalValue = value;
+                ValueChanged.InvokeAsync(value);
```

Comment:
I think you should check that the `value != Value` before raising the `ValueChanged`.
In addition, you should check that the user has requested to be subscribed via `if (ValueChanged.HasDelegate)`.